File: sdk/nodejs/src/model/system/data-source.ts



export const DataSourceResource = {
    resource: "DataSource",
    namespace: "system",
};

// Sub Types

// Resource Type
export interface DataSource {
    id: string;
version: number;
createdBy?: string;
updatedBy?: string;
createdOn?: string;
updatedOn?: string;
name: string;
description?: string;
backend: 'POSTGRESQL' | 'MYSQL' | 'MONGODB' | 'REDIS';
options: object;

}
// Resource and Property Names
export const DataSourceName = "DataSource";

export const DataSourceIdName = "Id";

export const DataSourceVersionName = "Version";

export const DataSourceCreatedByName = "CreatedBy";

export const DataSourceUpdatedByName = "UpdatedBy";

export const DataSourceCreatedOnName = "CreatedOn";

export const DataSourceUpdatedOnName = "UpdatedOn";

export const DataSourceNameName = "Name";

export const DataSourceDescriptionName = "Description";

export const DataSourceBackendName = "Backend";

export const DataSourceOptionsName = "Options";

File: sdk/python/apibrew/model/data_source.py
from datetime import datetime
from typing import List, Optional
from enum import Enum
from apibrew.entity import Entity, EntityInfo
from __future__ import annotations




class AuditData:
    createdBy: str
    updatedBy: str
    createdOn: datetime
    updatedOn: datetime


class Backend(Enum):
    POSTGRESQL = "POSTGRESQL"
    MYSQL = "MYSQL"
    MONGODB = "MONGODB"
    REDIS = "REDIS"



class DataSource:
    id: str
    version: int
    auditData: AuditData
    name: str
    description: str
    backend: Backend
    options: dict[str, str]

    @staticmethod
    def entity_info():
        return EntityInfo("system", "DataSource", "system-datasource")




